x/farming/keeper/plan.go

```diff
@@ -110,6 +110,39 @@ func (k Keeper) decodePlan(bz []byte) types.PlanI {
 	return acc
 }
 
+// GetNumPrivatePlans returns the current number of private plans.
+func (k Keeper) GetNumPrivatePlans(ctx sdk.Context) uint32 {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.NumPrivatePlansKey)
+	var val gogotypes.UInt32Value
+	k.cdc.MustUnmarshal(bz, &val)
+	return val.GetValue()
+}
+
+// SetNumPrivatePlans sets the current number of private plans.
+func (k Keeper) SetNumPrivatePlans(ctx sdk.Context, num uint32) {
+	store := ctx.KVStore(k.storeKey)
+	bz := k.cdc.MustMarshal(&gogotypes.UInt32Value{Value: num})
+	store.Set(types.NumPrivatePlansKey, bz)
+}
+
+// IncrementNumPrivatePlans increments the current number of private plans.
+func (k Keeper) IncrementNumPrivatePlans(ctx sdk.Context, amt uint32) {
+	num := k.GetNumPrivatePlans(ctx)
+	num += amt
+	k.SetNumPrivatePlans(ctx, num)
+}
+
+// DecrementNumPrivatePlans decrements the current number of private plans.
+func (k Keeper) DecrementNumPrivatePlans(ctx sdk.Context, amt uint32) {
+	num := k.GetNumPrivatePlans(ctx)
+	if num < amt { // sanity check
+		panic("cannot set negative NumPrivatePlans")
+	}
+	num -= amt
+	k.SetNumPrivatePlans(ctx, num)
+}
+
```

Comment:
I think it is over-engineering, the checking number of the private plan is only called when creating private plan logic, It could be charged iterating gas to private plan creator, GetNumPrivatePlans has a condition ( not terminated ) so It could increase the possibility of causing the mistake and invariant fail IMO

Reply:
Yes, I agree. Successfully removed.